docs/source/upgrading_your_components.md

```diff
@@ -0,0 +1,208 @@
+
+If using CouchDB as state database, there will be no `stateLeveldb` directory, as the state database data would stored within CouchDB instead. But similarly, if peer starts up and finds CouchDB databases are missing or at lower block height (based on using an older CouchDB backup), the state database will be automatically re-constructed to catch up to current block height. Therefore, if you backup peer ledger data and CouchDB data separately, ensure that the CouchDB backup is always older than the peer backup.
+
+## Prerequisites
+
+Before attempting to upgrade components, make sure you have all of the dependencies on your machine as described in [Prerequisites](./prereqs.html). This will not only ensure that you have the latest binaries, but the latest version of all dependencies.
+
+Also, make sure you have TLS enabled by issuing:
+
+```
+CORE_PEER_TLS_ENABLED=true
+```
+
+## Upgrade ordering nodes
+
+Orderer containers should be upgraded in a rolling fashion (one at a time). At a high level, the ordering node upgrade process goes as follows:
+
+1. Stop the ordering node.
+2. Back up the ordering node's ledger and MSP.
+3. Restart the ordering node with the latest images.
+
+Repeat this process for each node in your ordering service until the entire ordering service has been upgraded.
+
+### Set environment variables
+
+Export the following environment variables before attempting to upgrade your ordering nodes. Note that the use of `CORE_PEER` in these environment variables is not an error.
+
+* `ORDERER_ADDRESS`: the url of your ordering node. Note that you will need to export this variable for each node when upgrading it.
+* `LEDGERS_BACKUP`: the place in your local filesystem where you want to store the ledger being backed up. As you will see below, each node being backed up will have its own subfolder containing its ledger. You will need to create this folder.
+* `IMAGE_TAG`: The version you are upgrading to. This should have a prefix of `(go env GOARCH)`. For example `export IMAGE_TAG=$(go env GOARCH)-<version you are upgrading to>`.
+
+### Upgrade containers
+
+If you have a native deployment, you should copy your `orderer.yaml` file now. Later, you will need this file to compare against the `orderer.yaml` file that is bundled with the release.
+
+Let’s begin the upgrade process by **bringing down the orderer**:
+
+```
+docker stop $ORDERER_ADDRESS
+```
+
+Once the orderer is down, you'll want to **backup its ledger and MSP**:
+
+```
+docker cp $ORDERER_ADDRESS:/var/hyperledger/production/orderer/ ./$LEDGERS_BACKUP/$ORDERER_ADDRESS
+```
+
+Now **restart the ordering node** with our new image.
+
+For native deployments, this is where you will need to compare the `orderer.yaml` file bundled with the release against the `orderer.yaml` you copied earlier. A tool like `diff` can be helpful here. Once you have copied over the relevant differences, you can start the node:
+
+```
+peer node start
```

Comment:
@joealewine for orderers, command should be `orderer` instead of `peer node start`